api/app/Models/User.php

```diff
@@ -2,38 +2,62 @@
 
 namespace App\Models;
 
+use BeatSwitch\Lock\LockAware;
+use BeatSwitch\Lock\Callers\Caller;
 use Illuminate\Auth\Authenticatable;
 use Illuminate\Contracts\Auth\Authenticatable as AuthenticatableContract;
 
-class User extends BaseModel implements AuthenticatableContract
+class User extends BaseModel implements AuthenticatableContract, Caller
 {
-    use Authenticatable;
+    use Authenticatable, LockAware;
 
     const USER_TYPE_ADMIN = 'admin';
     const USER_TYPE_PUBLIC = 'public';
```

Comment:
@xiphiaz I'd think if using the guest convention, having `guest` as the base for any visitor, `normal` for the common registered user, and then finally the `admin` type. So `guest` would not be stored users.

Or are you thinking to use the user type `guest` for a common registered user? So just an update of the naming for that user type from public to guest?